frontend/apps/reader/modules/readersearch.lua

```diff
@@ -70,6 +70,7 @@ function ReaderSearch:onShowFulltextSearchInput()
 end
 
 function ReaderSearch:onShowSearchDialog(text, direction)
+    if text == '' then return end
```

Comment:
```suggestion
    if text == "" then return end
```

However, it should be in the callback, not here.

Reply:
```diff
diff --git a/frontend/apps/reader/modules/readersearch.lua b/frontend/apps/reader/modules/readersearch.lua
index 49d11041..ce81899b 100644
--- a/frontend/apps/reader/modules/readersearch.lua
+++ b/frontend/apps/reader/modules/readersearch.lua
@@ -50,6 +50,7 @@ function ReaderSearch:onShowFulltextSearchInput()
                 {
                     text = backward_text,
                     callback = function()
+                        if self.input_dialog:getInputText() == "" then return end
                         UIManager:close(self.input_dialog)
                         self:onShowSearchDialog(self.input_dialog:getInputText(), 1)
                     end,
@@ -58,6 +59,7 @@ function ReaderSearch:onShowFulltextSearchInput()
                     text = forward_text,
                     is_enter_default = true,
                     callback = function()
+                        if self.input_dialog:getInputText() == "" then return end
                         UIManager:close(self.input_dialog)
                         self:onShowSearchDialog(self.input_dialog:getInputText(), 0)
                     end,
```

Something like this. Or maybe as an enabled_func of some sort but that'd be slightly harder to add.

Reply:
Can you please point me to the reasons, why two conditions is better than one?

Reply:
The above patch won't close the input widget if it's empty and the user hits < or >. The user will have to hit Cancel to get out.
With your PR, it closes and no search is performed.
I think I'd prefer the above patch, which makes things more obvious - but I haven't checked how we do that in other input dialogs of this kind.

Reply:
As @poire-z said, two conditions isn't better than one but not closing the dialog is better than closing the dialog.

Reply:
That is, I was expressing a rough sketch of the desired behavior in the form of a patch, the actual patch could take a completely different form.

Reply:
> With your PR, it closes and no search is performed.

That is by design. This behaviour is the same as in the File search, Dictionary lookup, Wiki lookup.

Reply:
That behavior would be a *lack* of design. Apparently I never considered that scenario when I wrote (some of) those dialogs. :-) Closing it as if you clicked cancel is very unexpected.

Reply:
> Closing it as if you clicked cancel is very unexpected.

Can it be hardcored in the InputDialog to be applied everywhere?
(No reaction to the ok_button(s) if the input field is empty)

Reply:
In the sense of making that the default (rather than "hardcoded" as such), in general probably. But this particular specimen is a custom dialog iirc.

Reply:
And there are some other input dialog where entering/saving an empty string might make sense (ie Wikipedia languages, to reset it to its default of just using the book or UI language).
But given from this sample that it's the job of the caller to close or not close the dialog in its callback, the current behaviour of InputDialog not caring might be more flexible (and less brain work :)

Reply:
OK, what do you recommend right now?

Currently when pressing ok_button with empty input box
- File search, Dictionary lookup, Wiki lookup just close the input dialog
- Fulltext search tries to search and says No result, leaving search_dialog (arrows) open